has/haselements.go

```diff
@@ -0,0 +1,54 @@
+package has
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/corbym/gocrest"
+)
+
+// Checks wether each element of the array/slice matches each exptectation passed to ElementsWith
+// Panics if the actual is not an array/slice
+func ElementsWith(expects ...*gocrest.Matcher) *gocrest.Matcher {
```

Comment:
Good points!

for 1: I don't like the name `ArrayContaining` because go distinguishes between arrays and slices. Naming it `ArrayContaining` suggests it would just work for arrays. But I agree that `EveryElement` reads more intuitive.

for 2: You're right there. The implementation would be easier with one matcher. Combined with `AllOf` or `AnyOf` the same functionality can be provided as now.

for 3: I think we have two use cases here.

My implementation checks each element against all matchers. That has the advantage that for a slice of length n I can provide one matcher to check all of the items. The disadvantage is that the matcher has to be generic enough to match each Item.

Your suggested implementation has the advantage that the matcher can be more specific. But for a slice of length n, n matchers have to be provided. 

I am not sure but isn't there already an implementation like this in `is.ValueContaining`?

Thanks! :)

Reply:
`has.ValueContaining` currently only supports values and not matchers(it should probably also be in the `has` package.. but oh well). I have a wip to convert it to use matchers or values, but it would indeed have the effect of every value needing to be matched _in order_ with the _same length_.

>That has the advantage that for a slice of length n I can provide one matcher to check all of the items.

I cannot imagine a situation where you would have such a generic match that you would need all n items in the array or slice to match all n matchers passed in (apart from the base case where n = 1, which allOf would work nicely). Can you provide a real world example?

Reply:
Hmm. If I think about a real world example *in the context of a test* you're right. My last test cases all had slices with a length between one and three, which would be very capable with your suggestion.

Will provide an implementation.

Reply:
https://github.com/corbym/gocrest/pull/1/commits/60cb4492a0a3d45aae15b55fe0ed98032204bf72